sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1208,85 +1208,172 @@ class ColumnName(name: String) extends Column(name) {
    */
   def boolean: StructField = StructField(name, BooleanType)
 
+  /**
+   * Creates a new `StructField` of type boolean.
+   * @since 2.4.0
+   */
+  def boolean(nullable: Boolean): StructField = StructField(name, BooleanType, nullable)
```

Comment:
The NULL hints are not enforced. Thus, it is kind of risky to expose this to end users since it could generate a wrong result. We plan to ignore the user-specified NULL hints in the upcoming release.